app/src/mainwindow2.cpp

```diff
@@ -984,9 +984,16 @@ void MainWindow2::importGIF()
 void MainWindow2::lockWidgets(bool shouldLock)
 {
     QDockWidget::DockWidgetFeatures feat = shouldLock ? QDockWidget::NoDockWidgetFeatures : QDockWidget::AllDockWidgetFeatures;
+
     for (QDockWidget* d : mDockWidgets)
     {
         d->setFeatures(feat);
+
+        // https://doc.qt.io/qt-5/qdockwidget.html#setTitleBarWidget
+        // A empty QWidget looks like the tittle bar is hidden.
+        // nullptr means removing the custom title bar and restoring the default one
+        QWidget* customTitleBarWidget = shouldLock ? (new QWidget) : nullptr;
+        d->setTitleBarWidget(customTitleBarWidget);
```

Comment:
If we intend to add this to all our dock widgets, wouldn't it be better to apply this to the BaseDockWidget class instead?

Reply:
What do you mean exactly? you mean adding a function called `BaseDockWidget::hideTitlebar()`? 

I don't do so because it's one line code and won't be called anywhere except here.

Reply:
I thought again. Yea, probably we can do that to make code more expressive.